### PR TITLE
Comment feature api post

### PR DESCRIPTION
**Pull request by kennycason**
Enjoy the bad grammar and spelling :)

### CODE REVIEW

File: _posts/2015-09-23-api-endpoint-for-features.markdown [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+| Purchased Product | TRUE, FALSE | 1, -1 |
+| Mentions Price | TRUE, FALSE | 1, -1 |
+| Follower Count | (no categorical representation) | 0 - Long.MAX_VALUE |
+
+For most features, mapping to a set of categories and/or a numerical value is straight forward. Some features such as the continuously valued DataRank feature, [0.0, 100.0], does not have an obvious mapping to a set of discrete categories. To handle these types of discrete mappings, we provide a couple of options. One is to return the numerical value and let the client determine the categorical mappings. The second is that we partition the continuous value into five ranges, (VERY_LOW, LOW, MID, HIGH, VERY_HIGH).
+
+### From Comment Search to Comment Feature Search
+
+One of DataRank's strengths is our 40+ search filters that can be applied in combination to each of our search and analytic endpoints. The endpoint we decide to extend if the our primary comment search endpoint. Specifically, instead of returning the raw comments and meta data, we extended the endpoint to return only extract features. Having the extracted features only makes prepping the data for modeling/analysis that much easier.
```
Comment (aewhite) [inline _posts/2015-09-23-api-endpoint-for-features.markdown]: return only *extracted* features

**Comment by addamh** [comment]
Updated grammar and language issues directly on the PR instead of commenting. I believe you should include a conclusion at the end however instead of ending on code/examples.

**Comment by dempe** [comment]
Strongly agree with Addam's recommendation.

**Comment by kennycason** [comment]
Thanks a lot, I agree as well.

**Comment by kennycason** [comment]
@addamh @dempe @sawyercade I added a conclusion, I also shortened the code examples. I'm also wondering if I even really want the code examples. While I think they could add value because they show people very exact example problems that the data works well with, or if they distract from my main point. Thoughts?

**Comment by sawyercade** [comment]
Summarizing the discussion @kennycason and I had:

We should more closely match the ARFF format (commonly used among the ML community). This would entail one JSON object for each of the three ARFF sections: relation, attributes, and data.

The way the API returns data now, there's no way to know what the domain of a feature is. For example, if I return "2" as the value for a categorical field, I don't know how many possible values exist for that field. I also don't know what "2" indicates: it could indicate VERY_HIGH in a domain of [-2, 2], or it could be LOW in a domain of [0,10]. Using ARFF sections as a model for the JSON objects we return eliminates both the "what's the domain for this value?" question and the "what's the meaning/label of this value?" question.

**Pull request merged**